lib/src/json_schema/json_schema.dart

```diff
@@ -109,6 +111,31 @@ class JsonSchema {
         'Data provided to createSchema is not valid: Must be a Map (or bool in draft6 or later). | $data');
   }
 
+  /// Create a schema from a [Map].
+  ///
+  /// This method is syncronous, and supports reading all ref'd schema from a
+  /// [Map<String, JsonSchema>] for items, properties, and sub-properties of the root
+  /// schema. If you want to create a [JsonSchema] without having to fetch ref'd schemas up
+  /// front, and use [createSchemaWithProvidedRefs] instead.
+  ///
+  /// Typically the supplied [data] is result of [JSON.decode] on a JSON [String], and
+  /// the values on [providedRefs] are generated with more calls to [createSchemaWithProvidedRefs].
+  static JsonSchema createSchemaSync(dynamic data,
```

Comment:
change this to `createSchema` (make it the default), change the current method to `createSchemaAsync`